packages/compiler/src/core/checker.ts

```diff
@@ -5756,6 +5788,56 @@ export function createChecker(program: Program): Checker {
     );
   }
 
+  function arePropertiesAssignableToIndexer(
+    properties: Map<string, ModelProperty>,
+    indexerConstaint: Type,
+    diagnosticTarget: DiagnosticTarget,
+    relationCache: MultiKeyMap<[Type, Type], Related>
+  ): [Related, readonly Diagnostic[]] {
+    for (const prop of properties.values()) {
+      const [related, diagnostics] = isTypeAssignableToInternal(
+        prop.type,
+        indexerConstaint,
+        diagnosticTarget,
+        relationCache
+      );
+      if (!related) {
+        return [Related.false, diagnostics];
+      }
+    }
+
+    return [Related.true, []];
+  }
+
+  function areIndexAreCompatible(
```

Comment:
This needs a better name 😂

Reply:
any ideas?

- `hasIndexAndIsRelatedTo`
- `hasIndexAndIsSubTypeOf`
- `hasIndexAndIsAssignableTo`